lightly/data/_video.py

```diff
@@ -185,18 +185,25 @@ def _is_valid_file(filename):
 
         # keep track of valid files
         path = os.path.join(directory, fname)
-        instances.append(path)
+        video_instances.append(path)
 
     # get timestamps
     timestamps, fpss = [], []
-    for instance in instances:
+    for instance in video_instances[:]: # video_instances[:] creates a copy
 
         if AV_AVAILABLE and torchvision.get_video_backend() == 'pyav':
             # This is a hacky solution to estimate the timestamps.
             # When using the video_reader this approach fails because the 
             # estimated timestamps are not correct.
             with av.open(instance) as av_video:
                 stream = av_video.streams.video[0]
+
+                # check if we can extract the video duration
+                if not stream.duration:
+                    print(f'Video {instance} has no timestamp and will be skipped...')
+                    video_instances.remove(instance) # remove from original list (not copy)
```

Comment:
unsure how this works in python if instance is a reference and only the "list" was copied. This removes it correctly?